dspace-server-webapp/src/main/java/org/dspace/app/rest/scripts/handler/impl/RestDSpaceRunnableHandler.java

```diff
@@ -156,20 +157,48 @@ public void logInfo(String message) {
         String logMessage = getLogMessage(message);
         log.info(logMessage);
 
+        appendLogToProcess(message, ProcessLogLevel.INFO);
     }
 
     @Override
     public void logWarning(String message) {
         String logMessage = getLogMessage(message);
         log.warn(logMessage);
+        appendLogToProcess(message, ProcessLogLevel.WARNING);
     }
 
     @Override
     public void logError(String message) {
         String logMessage = getLogMessage(message);
         log.error(logMessage);
+        appendLogToProcess(message, ProcessLogLevel.ERROR);
     }
 
+
+    /**
+     * This method will ensure that the current {@link Process} has the given {@link org.dspace.scripts.ProcessLog}
+     * objects made and attached to it in the DB when a log is called.
+     * It'll use a separate Context for this and close this one immediately afterwards so that it's updated in
+     * real-time
+     * @param message   The message to be used in the log
+     * @param processLogLevel   The log level to be used in the log
+     */
+    private void appendLogToProcess(String message, ProcessLogLevel processLogLevel) {
+        Context context = new Context(Context.Mode.MANAGED);
```

Comment:
I'm worried about performance here.  Since you've setup the `MANAGED` context to always establish a *new* Hibernate Session (see above comment), it looks like you are going to open a new Hibernate Session (and DB connection) for every log statement.  

This seems very much like we are accidentally implementing the `session-per-operation` anti-pattern that Hibernate warns against:https://docs.jboss.org/hibernate/orm/5.0/userguide/en-US/html/ch06.html#session-per-operation   See also https://docs.jboss.org/hibernate/orm/4.3/manual/en-US/html/ch13.html#transactions-basics

We've tried very hard to avoid this anti-pattern so far, and we've been using the `session-per-request` pattern.